src/app/components/tfrs/tfrs.component.ts

```diff
@@ -0,0 +1,18 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'app-tfrs',
+  templateUrl: './tfrs.component.html',
+  styleUrls: ['./tfrs.component.scss']
+})
+export class TfrsComponent implements OnInit {
+  TfrId!: String | null;
```

Comment:
TFRId should be number as below
TfrId !: number;

Move all the logic to TFRComponent, and not here